tuplex/codegen/include/FunctionRegistry.h

```diff
@@ -101,6 +104,41 @@ namespace tuplex {
 
             SerializableValue createRandomChoiceCall(LambdaFunctionBuilder &lfb, llvm::IRBuilder<> &builder, const python::Type &argType, const SerializableValue &arg);
 
+            SerializableValue createIterCall(LambdaFunctionBuilder &lfb,
+                                             llvm::IRBuilder<>& builder,
+                                             const python::Type &argsType,
+                                             const python::Type &retType,
+                                             const std::vector<tuplex::codegen::SerializableValue> &args);
+
+            SerializableValue createNextCall(LambdaFunctionBuilder &lfb,
+                                             llvm::IRBuilder<>& builder,
+                                             const python::Type &argsType,
+                                             const python::Type &retType,
+                                             const std::vector<tuplex::codegen::SerializableValue> &args,
+                                             IteratorInfo *iteratorInfo);
+
+            SerializableValue createZipCall(LambdaFunctionBuilder &lfb,
+                                             llvm::IRBuilder<>& builder,
+                                             const python::Type &argsType,
+                                             const python::Type &retType,
+                                             const std::vector<tuplex::codegen::SerializableValue> &args,
+                                             IteratorInfo *IteratorInfo);
+
+            SerializableValue createEnumerateCall(LambdaFunctionBuilder &lfb,
+                                            llvm::IRBuilder<>& builder,
+                                            const python::Type &argsType,
+                                            const python::Type &retType,
+                                            const std::vector<tuplex::codegen::SerializableValue> &args,
+                                            IteratorInfo *iteratorInfo);
+
+            SerializableValue createIteratorRelatedSymbolCall(tuplex::codegen::LambdaFunctionBuilder &lfb,
```

Comment:
What does this function do? The others I get from the description. This one I am confused. Maybe add doxygen?

Reply:
got it, thanks.